docs/messages/README.md

```diff
@@ -17,33 +17,99 @@ The protocol comprises various messages, which can be classified as either _requ
 
-To verify the consumer identity, the **RESERVE_FLAVOR** message must also include an authentication token that will be then validated by a Trusted third-party authentication and authorization service (TODO: detail here the vPresentation to authenticate the consumer).
-Once received, the provider checks if the flavor is still available (there might be some delay between the **LIST_FLAVOR** message and the subsequent reservation). and if so it replies with a summary of the reservation process including the _TransactionID_ and the _Time To Purchase (TTP)_, i.e., the time by which the Flavor must be purchased. 
+Similarly to the **LIST_FLAVOR** message, the **RESERVE_FLAVOR** message must also include the VP of the consumer.
+Once received, the provider (i) verifies the VP of the consumer, and (ii) checks if the flavor is still available (there might be some delay between the **LIST_FLAVOR** message and the subsequent reservation).
+If the two requirements are satisfied it replies with a summary of the reservation process including the _TransactionID_ and the _Time To Purchase (TTP)_, i.e., the time by which the Flavor must be purchased, and, again witht its VP (the Figure needs to be updated). 
 This allows reserved Flavors to be released in case either the consumer becomes unreachable, or the subsequent purchase process exceeds a predefined threshold.
 If the Flavor is not available a 404 error message is sent to the consumer.
 
 ## PURCHASE A FLAVOR
 
 The **PURCHASE_FLAVOR** message is sent by the consumer upon receipt of the provider's response during the reservation phase to complete the purchase of an offered flavor.
 Specifically, the **PURCHASE_FLAVOR** message notifies the intention of the consumer to finalize the purchase of the resource.
-To do so, the consumer sends the **PURCHASE_FLAVOR** message including the _TransactionID_ (obtained with the **RESERVE_FLAVOR**) and the identification token to the provider (TODO: detail here the vPresentation to authenticate the consumer).
+To do so, the consumer sends the **PURCHASE_FLAVOR** message including the _TransactionID_ (obtained with the **RESERVE_FLAVOR**) and the VP to the provider.
 If authorized, the consumer will then be prompted to a payment service (either external or managed by the provider) and, if successful, a copy of the _Contract_ is returned to the consumer, detailing the purchase and the information required to access the purchased resource (e.g., IP address, API endpoint).
 
+To ensure the contract wasn't changed along the wayand verify that the provider is who it says it is, the contract is signed using a **JSON Web Token (JWT)**.
+Specifically, the contract is first signed by the provider, and then returned to the consumer, that, in turn, can sing it and store in in a DLT.
+AS a result, the contract signed from both parties bond both the consumer and the producer to the agreed upon transaction.
```

Comment:
Hi Stefano,

 I think we could improve the clarity and accuracy of this part with this revised version. How about replace or change what you consider of the current with this:
 
To ensure the contract hasn't been changed and to verify the provider's identity, the contract is encapsulated within a JSON Web Token (JWT) and signed using the DID private key. Specifically, the contract is first signed by the provider using the JWT, then returned to the consumer, who can add their own signature to the JWT. More nested signatures can be added as needed.

Hashes of the contracts can be stored in the Distributed Ledger Technology (DLT), ensuring immutability and traceability. By retrieving the contract's hash from the DLT, we can verify the contract's existence and integrity, providing notary-like effects. The signatures on the JWT can be verified using the public key associated with the DID of the signer, which can be publicly accessed in the DLT. This process bonds both the consumer and the producer to the agreed-upon transaction and ensures the integrity and authenticity of the contract.